### PR TITLE
pickList: tests for api methods, callbacks, options added

### PR DESCRIPTION
**Pull request by jhuska**
Please review before merging.

Mainly the test for `disable` option. Please see its comments. It has weird behavior.

### CODE REVIEW

File: test/widgets/select/pick-list/pick-list-options.js [hunk 1]
```diff
@@ -261,5 +261,83 @@ define(['widget-test-base', 'jquery', 'jquery-ui', 'src/widgets/select/orderingL
         expect(getColumnClass(row_1_cells, 2)).not.toMatch(newColumnClasses[1]);
       });
     });
+
+    it('disabled option: ', function () {
```
Comment (jhuska) [inline test/widgets/select/pick-list/pick-list-options.js:265]: This is the weird test. By setting `disable` option to true it is not disabled.

When playing with it in the demo, the same setting makes `pickList` disabled.

File: test/widgets/select/pick-list/pick-list-source.html [hunk 2]
```diff
@@ -231,9 +245,19 @@
...
+                    <td>Item 10</td>
+                    <td class="last">10</td>
+                  </tr>
+                  <tr data-key="11" class="ui-selectee">
+                    <td class="first"><i class="icon-trash"></i></td>
+                    <td>Item 11</td>
+                    <td class="last">11</td>
                   </tr>
-                  <tr data-key="9" class="fill-item" style="height: 88px; display: table-row; ">
+                  <tr data-key="9" class="fill-item" style="height: 26px; display: table-row; ">
```
Comment (jhuska) [inline test/widgets/select/pick-list/pick-list-source.html]: This element style is the cause of the DOM failure.

I am not sure why that element is there ? It is not displayed and it is there even when the list is empty. Is that a bug ?

Comment (lfryc) [inline test/widgets/select/pick-list/pick-list-source.html]: This is tracked here: https://github.com/richwidgets/richwidgets/issues/41

If you have any ideas for resolution of #41, let's discuss there.

**Comment by lfryc** [comment]
Hey Juraj, these tests fail:

```
 PhantomJS 1.9.2 (Linux).widget(pickList): options:.disabled option 
 PhantomJS 1.9.2 (Linux).widget(pickList): source.pickList from <table> markup  
```

**Comment by lfryc** [comment]
Please could you also rebase to master? The orderingList/pickList resourcws were renamed there to pick-list/ordering-list

File: test/widgets/select/pick-list/pick-list-options.js [hunk 3]
```diff
@@ -261,5 +261,78 @@ define(['widget-test-base', 'jquery', 'jquery-ui', 'src/widgets/select/orderingL
...
+        // var changeCallback = jasmine.createSpy('changeCallback');
+
+        var options = {
+          disabled: true,
+          // change: changeCallback
+        };
+        // when
+        element.pickList(options);
+        
+        runs(function () {
```
Comment (bleathem) [inline test/widgets/select/pick-list/pick-list-options.js]: Get rid of the runs() method here.  This is not an async test.

File: test/widgets/select/pick-list/pick-list-options.js [hunk 4]
```diff
@@ -261,5 +261,78 @@ define(['widget-test-base', 'jquery', 'jquery-ui', 'src/widgets/select/orderingL
...
+        // });
+
+        // runs(function () {
+        //   expect(changeCallback).not.toHaveBeenCalled();
+        // });
+
+        options = {
+          disabled: false,
+        };
+        element.pickList(options);
```
Comment (bleathem) [inline test/widgets/select/pick-list/pick-list-options.js]: you have to destroy the widget before instantiating it again.

Alternatively use the `element.pickList('option', 'disabled', false);` syntax for mutating options (depends what you are trying to test)

**Comment by jhuska** [comment]
I have updated the commit according to your comments.

**Comment by bleathem** [comment]
This looks good to me!

**Comment by lfryc** [comment]
Merged: https://github.com/richwidgets/richwidgets/commit/a48346a2511d6d66f6dc82f0d9b79cbde8f16c5f

**Pull request closed without merging**